### PR TITLE
all: logging: Remove log_strdup function

### PR DESCRIPTION
**Pull request by nordic-krch**
Logging v1 has been removed and log_strdup wrapper function is no
longer needed. Removing the function and its use in the tree.

Part of #46500.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Review by PavelVPV — DISMISSED** [review]

**Comment by nordic-krch** [comment]
CI is somehow green except for false compliance check which does not like word `set` in the log message:
```
-:[25](https://github.com/zephyrproject-rtos/zephyr/runs/6978715279?check_suite_focus=true#step:9:26)47: WARNING:EMBEDDED_FUNCTION_NAME: Prefer using '"%s...", __func__' to using 'set', this function's name, in a string
#2547: FILE: subsys/bluetooth/host/settings.c:181:
+			BT_DBG("Name set to %s", bt_dev.name);
```

**Review by andyross — DISMISSED** [review]
The extra fixups might want to be submitted elsewhere and first, but that seems minor.

**Review by dcpleung — DISMISSED** [review]
If you need to rebase, it will be a good idea to move the west manifest commit earlier to maintain bisect-ibility.

**Comment by nordic-krch** [comment]
switched commit order as @dcpleung suggested. @dcpleung @andyross can you reapprove?

**Review by dcpleung — DISMISSED** [review]

**Review by ycsin — DISMISSED** [review]

**Review by andyross — DISMISSED** [review]
Refresh +1

**Review by aasthagr — DISMISSED** [review]

**Review by nvlsianpu — DISMISSED** [review]

**Review by nvlsianpu — DISMISSED** [review]

**Comment by carlescufi** [comment]
@PavelVPV can you re-review please?

**Review by carlescufi — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

**Review by jfischer-no — DISMISSED** [review]

**Review by anangl — APPROVED** [review]

File: west.yml [hunk 1]
```diff
@@ -161,7 +161,7 @@ manifest:
       revision: 12019623bbad9eb54fe51066847a7cbd4b4eac57
       path: modules/lib/loramac-node
     - name: lvgl
-      revision: df717ac87a9fd80246cc8df24554475e59bda21f
+      revision: 892820523cfdd5dbf6d57b1fb4e7d0c0b573b94f
```
Comment (anangl) [inline west.yml:164]: This should be 07eab463f991f9ea1738a671fa81ad1e54f50ffb. 892820523cfdd5dbf6d57b1fb4e7d0c0b573b94f is the commit SHA from that lvgl PR.

**Review by anangl — CHANGES_REQUESTED** [review]

File: west.yml (hunk 1 above)
Comment (nordic-krch) [reply west.yml]: done

**Review by anangl — APPROVED** [review]

**Review by ycsin — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**